collect_app/src/main/java/org/odk/collect/android/utilities/SharedPreferencesUtils.java

```diff
@@ -0,0 +1,281 @@
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_PROTOCOL;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SERVER_URL;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SHOW_SPLASH;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SUBMISSION_URL;
+
+/**
+ * Created by shobhit on 12/4/17.
+ */
+
+public class SharedPreferencesUtils {
+
+    private final Context context = Collect.getInstance();
+    private final SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(context);
+    private final SharedPreferences.Editor editor = sharedPrefs.edit();
+    private final SharedPreferences adminSharedPrefs =
+            Collect.getInstance().getSharedPreferences(ADMIN_PREFERENCES, MODE_PRIVATE);
+    private final SharedPreferences.Editor adminEditor = adminSharedPrefs.edit();
+
+    String getJSONFromPreferences(Collection<String> keys) throws JSONException {
+        SharedPreferencesUtils obj = new SharedPreferencesUtils();
+        keys.addAll(ALL_GENERAL_KEYS);
+        JSONObject sharedPrefJson = obj.getModifiedPrefs(keys);
+        Timber.i(sharedPrefJson.toString());
+        return sharedPrefJson.toString();
+    }
+
+    private JSONObject getModifiedPrefs(Collection<String> keys) throws JSONException {
+        JSONObject prefs = new JSONObject();
+        JSONObject adminPrefs = new JSONObject();
+        JSONObject generalPrefs = new JSONObject();
+
+        for (String key : keys) {
+            String stringValue;
+            String defaultStringValue;
+
+            // checking for admin password
+            if (key.equals(KEY_ADMIN_PW)) {
+                stringValue = adminSharedPrefs.getString(key, "");
+                if (!stringValue.equals("")) {
+                    adminPrefs.put(key, stringValue);
+
+                    //skip further checking
+                    continue;
+                }
+            }
+
+            try {
+                stringValue = getStringValue(key);
+                defaultStringValue = getDefaultStringValue(key);
+                if (!stringValue.equals(defaultStringValue)) {
+                    generalPrefs.put(key, stringValue);
+                }
+            } catch (ClassCastException e) {
+                try {
+                    boolean booleanValue = getBooleanValue(key);
+                    boolean defaultBooleanValue = getDefaultBooleanValue(key);
+                    if (booleanValue != defaultBooleanValue) {
+                        generalPrefs.put(key, booleanValue);
+                    }
+                } catch (ClassCastException e1) {
+                    long longValue = getLongValue(key);
+                    long defaultLongValue = getDefaultLongValue(key);
+                    if (longValue != defaultLongValue) {
+                        generalPrefs.put(key, longValue);
+                    }
+                }
+            }
```

Comment:
Yes, you are absolutely right. I also agree that this is not the best approach to implement this. But I plan to refactor the sharedPrefs in a separate PR unless you disagree with me and feel like this should be addressed in this PR itself